nf_core/pipeline-template/{{cookiecutter.name_noslash}}/bin/check_samplesheet.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import errno
+import argparse
+
+def parse_args(args=None):
+    Description = 'Reformat {{ cookiecutter.name }} samplesheet file and check its contents.'
+    Epilog = """Example usage: python check_samplesheet.py <FILE_IN> <FILE_OUT>"""
+
+    parser = argparse.ArgumentParser(description=Description, epilog=Epilog)
+    parser.add_argument('FILE_IN', help="Input samplesheet file.")
+    parser.add_argument('FILE_OUT', help="Output file.")
+    return parser.parse_args(args)
+
+
+def make_dir(path):
+    if not len(path) == 0:
+        try:
+            os.makedirs(path)
+        except OSError as exception:
+            if exception.errno != errno.EEXIST:
+                raise
+
+
+def print_error(error,line):
+    print("ERROR: Please check samplesheet -> {}\nLine: '{}'".format(error,line.strip()))
+    sys.exit(1)
```

Comment:
Very minor indeed, but a nicer way to do this (at least, a pattern I'm using more myself recently) is to raise an exception when you hit a problem. You can then capture this at the level calling the function and print the error message there instead.

Functionally makes very little difference, except that it's easier to reuse the function elsewhere in larger codebases.

Reply:
Yeah, not sure what to do about this to be honest 😅 Happy for you to sneak an `exception` in though.

Reply:
No need to do anything, it's a small script so not really much point. Just something to file away in the cupboards for your next python project 😄